### PR TITLE
Command-line output in JSON format

### PR DESCRIPTION
**Pull request by Zettat123**
At present, the command output is in a human-readable format, making it difficult for programs to parse it. This PR introduces a new global command-line option `output-json`, enabling commands to showcase outputs in JSON format so that programs can parse the output more easily.

There are a range of different commands in Gitea and making them support JSON format is a big job. In this PR, I have focused on enhancing the `gitea admin user list` command. Subsequently, we can iteratively enhance other commands to support JSON output.

### CODE REVIEW

File: cmd/admin_user_list.go [hunk 1]
```diff
@@ -33,19 +33,59 @@ func runListUsers(c *cli.Context) error {
...
+			userInfos = append(userInfos, &userInfo{
+				ID:       user.ID,
+				Name:     user.Name,
+				Email:    user.Email,
+				IsActive: user.IsActive,
+				IsAdmin:  user.IsAdmin,
+			})
+		}
+
+		if c.IsSet("admin") {
```
Comment (lunny) [inline cmd/admin_user_list.go:73]: Why we need to detect if the flag has been set?

Comment (Zettat123) [reply cmd/admin_user_list.go:73]: 2FA will only be set if "admin" is not set. I keep the code behavior the same as before.

https://github.com/go-gitea/gitea/blob/e08f1a9cbd582c73918e401eeba36261627f44a7/cmd/admin_user_list.go#L43-L56

**Review by lunny — APPROVED** [review]

File: cmd/main.go [hunk 2]
```diff
@@ -77,6 +77,10 @@ func appGlobalFlags() []cli.Flag {
 			Aliases: []string{"w"},
 			Usage:   "Set Gitea's working path (defaults to the Gitea's binary directory)",
 		},
+		&cli.BoolFlag{
+			Name:  "output-json",
+			Usage: "Output in JSON format (not all commands are supported)",
+		},
```
Comment (wxiaoguang) [inline cmd/main.go:83]: It's not right to put it into global flags.

**Comment by wxiaoguang** [comment]
> There are a range of different commands in Gitea and making them support JSON format is a big job. In this PR, I have focused on enhancing the `gitea admin user list` command. Subsequently, we can iteratively enhance other commands to support JSON output.

Why not call the API directly?

**Review by wxiaoguang — CHANGES_REQUESTED** [review]
I do not see much benefit to introduce "JSON output" for CLI at the moment.

Most tasks could be done by API, right? If yes, the API system could be improved, because the API is more stable, and better documented.

**Comment by techknowlogick** [comment]
closing per above feedback

**Pull request closed without merging**